community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -240,16 +245,16 @@ public void failure()
     @Override
     public boolean shouldBeTerminated()
     {
-        return terminated;
+        return termination.shouldBeTerminated();
```

Comment:
Yeah, this would be an indirection. It is probably not a bad idea for KTI to implement `TxTermination` directly. I'll do this.